skiko/src/awtMain/objectiveC/macos/DisplayLinkThrottler.mm

```diff
@@ -88,7 +88,14 @@ - (instancetype)initWithWindow:(NSWindow *)window {
             [weakSelf onScreenDidChange];
         }];
 
-        [self onScreenDidChange];
+        if (NSThread.currentThread.isMainThread) {
+            [self onScreenDidChange];
+        } else {
+            // In case of OpenJDK, EDT thread != NSThread main thread
+            dispatch_async(dispatch_get_main_queue(), ^{
+                [self onScreenDidChange];
```

Comment:
```suggestion
                [weakSelf onScreenDidChange];
```
In case the layer is created and right away disposed (there are cases when this can happen).

Note: I don't have knowledge of Objective C weak references, so I can't be sure if this is correct. The point is to not call the callback after the object is deallocated.